index.js

```diff
@@ -326,6 +326,7 @@
    */
 
   Assertion.prototype.match = function (regexp) {
+    if (!regexp.exec) regexp = new RegExp(regexp);
```

Comment:
> I would argue the new
> String cases are a) uncommon b) not as important as cases such as {exec: 1}

Agree that `new String` is uncommon. Which is really besides the point since my objection is primarily on the duck-typing grounds.

> Doesn't `new RegExp` call toString implicitly

Yep, I guess it does (chrome anyway):
```
new RegExp({toString: function(){ return "HI" }})
/HI/
```

So that means `RegExp(regexp.toString())` is definitely out. Leaving us back to the original form?